File: CS/2149_Rearrange_Array_Elements_by_Sign.cs
public class Solution {
    public int[] RearrangeArray(int[] nums) {
        int[] result = new int[nums.Length];
        int p = 0;
        int n = 1;
        for(int i = 0; i < nums.Length; i++){
            if(nums[i] > 0){
                result[p] = nums[i];
                p += 2;
            }
            else{
                result[n] = nums[i];
                n += 2;
            }
        }
        return result;
    }
}

File: Java/2149_Rearrange_Array_Elements_by_Sign.java
class Solution {
    public int[] rearrangeArray(int[] nums) {
        int[] result = new int[nums.length];
        List<Integer> positive = new ArrayList<>();
        List<Integer> negative = new ArrayList<>();
        for(int i = 0; i < nums.length; i++){
            if(nums[i] > 0){
                positive.add(nums[i]);
            }
            else{
                negative.add(nums[i]);
            }
        }
        for(int i = 0; i < positive.size(); i++){
            result[2*i] = positive.get(i);
            result[2*i + 1] = negative.get(i);
        }
        return result;
    }
}
